__tests__/app.test.js

```diff
@@ -24,30 +39,73 @@ describe("Topics API", () => {
 				});
 		});
 	});
-});
-describe("Endpoints API", () => {
-	describe("Get Request", () => {
-		test("endpoints.json formatting correct", () => {
-			expect(endpoints).toBeInstanceOf(Object);
+	describe("Articles Endpoint", () => {
+		test("Get: 200, should return an array of all articles", () => {
+			return request(app)
+				.get("/api/articles")
+				.expect(200)
+				.then(({ body }) => {
+					const { articles } = body;
+					console.log(articles);
+					expect(articles).toBeInstanceOf(Array);
+					expect(articles.length).not.toBeLessThan(1);
+					expect(articles).toBeSortedBy("created_at", { descending: true });
+					articles.forEach((article) => {
+						expect(article).toHaveProperty("author");
+						expect(article).toHaveProperty("title");
+						expect(article).toHaveProperty("article_id");
+						expect(article).toHaveProperty("topic");
+						expect(article).toHaveProperty("created_at");
+						expect(article).toHaveProperty("votes");
+						expect(article).toHaveProperty("article_img_url");
+						expect(article).toHaveProperty("comment_count");
+						expect(article).not.toHaveProperty("body");
+					});
+				});
 		});
-		test("Get: 200, should return a list of endpoints from json file", () => {
+		test("Get: 200, should return a single article", () => {
 			return request(app)
-				.get("/api")
+				.get("/api/articles/3")
 				.expect(200)
 				.then(({ body }) => {
-					const endpointList = body.endpoints;
-					expect(endpointList).toEqual(endpoints);
-					expect(endpointList).not.toBe(endpoints);
+					const { article } = body;
+					expect(article).toBeInstanceOf(Array);
+					expect(article.length).not.toBeLessThan(1);
+					article.forEach((article) => {
+						expect(article.article_id).toBe(3);
+						expect(article).toHaveProperty("author");
+						expect(article).toHaveProperty("title");
+						expect(article).toHaveProperty("article_id");
+						expect(article).toHaveProperty("body");
+						expect(article).toHaveProperty("topic");
+						expect(article).toHaveProperty("created_at");
+						expect(article).toHaveProperty("votes");
+						expect(article).toHaveProperty("article_img_url");
+					});
 				});
 		});
 	});
 });
 
-// Implement tests later when more methods have been implemented
 describe("Error Testing", () => {
-	test("Get: 404, Returns error 404 when given an invalid url", () => {
-		return request(app).get("/api/topic").expect(404);
+	describe("Endpoint errors", () => {
```

Comment:
I know it's not for this ticket, but I just noticed this test and the one on line 101 shouldn't be in their own describe blocks. They are valid error tests, but should be within your describe block for ticket 4. 🙂